drivers/pwm/pwm_nrf5_sw.c

```diff
@@ -12,6 +12,8 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(pwm_nrf5_sw);
 
+#define PWM_0_MAP_SIZE DT_NORDIC_NRF_SW_PWM_0_CHANNEL_COUNT
```

Comment:
Maybe it would be good to check here if the number of channels specified in DTS is not too high?
Though the check would not look very nice:
```
/* One compare channel is needed to set the PWM period, hence +1. */
#if ((DT_NORDIC_NRF_SW_PWM_0_CHANNEL_COUNT + 1) > \
     (_CONCAT(_CONCAT(TIMER, DT_NORDIC_NRF_SW_PWM_0_TIMER_INSTANCE), _CC_NUM)))
```

Reply:
I think this is a good idea - at least the user will be notified that his parameter is wrongly set, instead of just running with it and failing "silently".

Doesnt the check ....CHANNEL_COUNT < ...._CC_NUM look better? We just need to be less than number of CC, we dont care if the user isnt using all of them?
If we do +1, it needs to be equal instead of > i believe.

Reply:
> Doesnt the check ....CHANNEL_COUNT < ...._CC_NUM look better? We just need to be less than number of CC, we dont care if the user isnt using all of them?
> If we do +1, it needs to be equal instead of > i believe.

My intention was to place an `#error` inside this check.

Reply:
Yes, that makes sense (Y)

Reply:
Included. Will update PR soon

Reply:
/* One compare channel is needed to set the PWM period, hence +1. */
#define PWM_0_MAP_SIZE DT_NORDIC_NRF_SW_PWM_0_CHANNEL_COUNT
#if ((DT_NORDIC_NRF_SW_PWM_0_CHANNEL_COUNT + 1) > \
	 (_CONCAT(_CONCAT(TIMER, DT_NORDIC_NRF_SW_PWM_0_TIMER_INSTANCE), _CC_NUM)))
#error "PWM channel-count must be one less than timer compare channels count."
#endif

Reply:
Actually, it doesn't have to be one less then CC count. It can be lower.
What about just "Invalid number of PWM channels configured."?

Reply:
Agreed.